tests/v2/validation/fleet/provisioning/airgap/fleet_in_airgap_test.go

```diff
@@ -0,0 +1,224 @@
+
+	user := &management.User{
+		Username: testuser,
+		Password: testpassword,
+		Name:     testuser,
+		Enabled:  &enabled,
+	}
+
+	newUser, err := users.CreateUserWithRole(client, user, "user")
+	require.NoError(a.T(), err)
+
+	standardUserClient, err := client.AsUser(newUser)
+	require.NoError(a.T(), err)
+
+	a.standardClient = standardUserClient
+
+	corralConfig := corral.Configurations()
+	err = corral.SetupCorralConfig(corralConfig.CorralConfigVars, corralConfig.CorralConfigUser, corralConfig.CorralSSHPath)
+	require.NoError(a.T(), err)
+
+	a.corralPackage = corral.PackagesConfig()
+
+	bastionIP := corralConfig.CorralConfigVars[corralBastionIP]
+	bastionUser := corralConfig.CorralConfigVars[corralSSHUser]
+
+	// format the privateKey string for use with ssh package.
+	privateKey := corralConfig.CorralConfigVars[corralPrivateKey]
+	privateKey = strings.Replace(privateKey, "\\n", "\n", -1)
+	privateKey = strings.Replace(privateKey, "\"", "", -1)
+	privateKey = fmt.Sprintf("%s", privateKey)
+
+	err = corral.UpdateCorralConfig(corralBastionIP, bastionIP)
+	require.NoError(a.T(), err)
+
+	registrySetting, err := client.Management.Setting.ByID(systemRegistry)
+	require.NoError(a.T(), err)
+
+	a.registryFQDN = registrySetting.Value
+
+	internalIP, err := setupAirgapFleetResources(bastionUser, bastionIP, privateKey)
+	require.NoError(a.T(), err, "failed to setup local git repo on bastion")
+
+	fleetSecretName, err := createFleetSSHSecret(client, privateKey)
+	require.NoError(a.T(), err, "failed to create SSH Secrets for fleet")
+
+	a.fleetGitRepo = &v1alpha1.GitRepo{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fleet.FleetMetaName + namegenerator.RandStringLower(5),
+			Namespace: fleet.Namespace,
+		},
+		Spec: v1alpha1.GitRepoSpec{
+			Repo:            fmt.Sprintf("%s@%s:/home/%s/%s", bastionUser, internalIP, bastionUser, fleetExampleFolderName),
+			Branch:          fleet.BranchName,
+			Paths:           []string{fleet.GitRepoPathLinux},
+			CorrectDrift:    &v1alpha1.CorrectDrift{},
+			ImageScanCommit: v1alpha1.CommitSpec{AuthorName: "", AuthorEmail: ""},
+			Targets: []v1alpha1.GitTarget{
+				{
+					ClusterSelector: &metav1.LabelSelector{
+						MatchExpressions: []metav1.LabelSelectorRequirement{
+							{
+								Key:      fleet.MatchKey,
+								Operator: fleet.MatchOperator,
+								Values: []string{
+									fleet.HarvesterName,
+								},
+							},
+						},
+					},
+				},
+			},
+			ClientSecretName: fleetSecretName,
+		},
+	}
+
+	a.client, err = a.client.ReLogin()
+	require.NoError(a.T(), err)
+
+	if a.clustersConfig.RKE2KubernetesVersions == nil {
+		rke2Versions, err := kubernetesversions.ListRKE2AllVersions(a.client)
+		require.NoError(a.T(), err)
+
+		a.clustersConfig.RKE2KubernetesVersions = []string{rke2Versions[len(rke2Versions)-1]}
+	}
+
+	if a.clustersConfig.CNIs == nil {
+		a.clustersConfig.CNIs = []string{fleet.CniCalico}
+	}
+}
+
+func (a *AirGapRKE2CustomClusterTestSuite) TestCustomClusterWithGitRepo() {
```

Comment:
if this is specific to RKE2, I would suggest we add RKE2 as well in the name so its clear that its applicable for rke2 only.

Reply:
same answer as above; cluster type isn't important for testing provisioning + fleet.

Reply:
It might make sense to add a comment expressing this in case someone in the future is looking at this test.

Reply:
is this sufficient @floatingman ? https://github.com/rancher/rancher/pull/47662/files#diff-55a941deb6ea319a6dad52528959cb64fcd25f679c50895883a4173f874e271dR7